scripts/gee

```diff
@@ -4836,6 +4836,82 @@ function gee__restore_all_branches() {
   _info "Done."
 }
 
+##########################################################################
+# recover_stashes
+##########################################################################
+
+_register_help "recover_stashes" "Recover identifiers for old, dropped stashes." \
+<<'EOT'
+Usage: recover_stashes <"optional string to search for">
+
+The `recover_stashes` command is useful if you accidentally dropped some
+stashed changes that you wanted to hold onto.  Even dropped stashes
+can be recovered if you know the right hash for the stash, and this
+command helps you discover them.
+
+By default, `recover_stashes` will print the hash ids for all stashes
+in your repo, sorted by author-date.  Any additional options are used
+to filter the commits (ie. by filename or file contents).
+
+Results are reported as a list of commits and their author-dates.
+
+For example:
+
+    $ gee recover_stashes enkit
+    Created temporary FSCKFILE=/tmp/stashes-jonathan.0JTAXI
+    Searching your repo for discarded stash commits...
+    Checking object directories: 100% (256/256), done.
+    Checking objects: 100% (8266/8266), done.
+    Cached search results to FSCKFILE=/tmp/stashes-jonathan.0JTAXI
+    Found 58 commits.
+    Filtering candidate commits...
+    Found 4 matches.
+      914c9f7f2a43e34c8b1f8ac26f478d4a3ae66c46 Sat Sep 24 12:56:57 2022 -0700
+      7b81aec0ed7806963e79bd2b236523aba57405a7 Tue Oct 4 07:33:03 2022 +0000
+      b957986e8044e8b3097591bae25dfee04e567469 Tue Oct 4 09:32:05 2022 -0700
+      184e791a27fe0f2dd3a7c87a7a9ea4c384d498e7 Thu Oct 6 06:01:44 2022 +0000
+    Remove /tmp/stashes-jonathan.0JTAXI if you no longer need it.
+
+EOT
+
+function gee__recover_stashes() {
+  local SEARCH_FOR="$*"
+
+  if [[ -z "${FSCKFILE}"  ]]; then
+    FSCKFILE="$(mktemp /tmp/stashes-"$USER".XXXXXX)"
+    _info "Created temporary FSCKFILE=${FSCKFILE}"
+  fi
+
+  if [[ ! -s "${FSCKFILE}" ]]; then
+    _info "Searching your repo for discarded stash commits..."
+    git fsck --no-reflog | awk '/dangling commit/ {print $3}' > "${FSCKFILE}"
+    _info "Cached search results to FSCKFILE=${FSCKFILE}"
+  fi
+
+  local -a COMMITS
+  mapfile -t COMMITS < "${FSCKFILE}"
+  echo "Found ${#COMMITS[*]} commits."
+
+  _info "Filtering candidate commits..."
+  local -a MATCHES=()
+  i=0
+  while (( i < ${#COMMITS[@]} )); do
+    commit="${COMMITS[$i]}"
+    # echo "Trying commit $((i+1))/${#COMMITS[@]}: ${commit}"
+    if grep -q "${SEARCH_FOR}" < <(git show "${commit}" || /bin/true); then
```

Comment:
Out of curiosity: why the `<(...)` instead of a pipe `git show | grep`?

Reply:
I was starting to do something fancy, and ended up doing something simple.  `git | grep` is better.